packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts

```diff
@@ -1000,3 +1123,14 @@ function unwrapSafeUrl(value: string|SafeValue): string {
   }
   return unwrapSafeValue(value);
 }
+
+// Transform function to handle inputs which may be booleans, strings, or string representations
+// of boolean values. Used for the placeholder attribute.
+export function booleanOrStringAttribute(value: boolean|string): boolean|string {
+  if (typeof value === 'boolean') {
+    return value;
+  } else if (value === 'true' || value === 'false') {
+    return Boolean(value);
+  }
+  return value;
```

Comment:
What do you think of reusing the Angular API for all cases except the "data:" case? So we're not re-creating the logic unnecessarily

```suggestion
  if (value.startsWith('data:') {
    return value;
  } else  {
    return booleanAttribute(value);
  }
```

Reply:
Yeah, this is better. Adopted!